sentence_transformers/sparse_encoder/models/SpladePooling.py

```diff
@@ -45,32 +55,77 @@ def __init__(
         if activation_function not in self.SPLADE_ACTIVATION:
             raise ValueError("activation_function must be either 'relu' or 'log1p_relu'")
         self.word_embedding_dimension = word_embedding_dimension  # This will be set in the forward method
+        self.chunk_size = chunk_size
 
-    def forward(self, features: dict[str, torch.Tensor]) -> dict[str, torch.Tensor]:
-        """Forward pass of the model.
+    def forward(
+        self,
+        features: dict[str, torch.Tensor],
+    ) -> dict[str, torch.Tensor]:
+        """
+        Forward pass of the model.
         Args:
-            features: Dictionary containing input features with 'token_embeddings' key as MLM logits.
+            features: Dictionary containing input features. Expects:
+                - 'token_embeddings': MLM logits (shape: batch_size, seq_length, vocab_size).
+                - 'attention_mask': Attention mask (shape: batch_size, seq_length).
         Returns:
             Dictionary containing SPLADE pooled embeddings
         """
-        # Get the MLM head logits (shape: batch_size, seq_length, vocab_size)
         mlm_logits = features["token_embeddings"]
+        attention_mask = features["attention_mask"]  # Shape: [batch_size, seq_length]
 
-        # Apply ReLU and log transformation for SPLADE
-        if self.activation_function == "relu":
-            splade_scores = torch.log1p(torch.relu(mlm_logits))
-        elif self.activation_function == "log1p_relu":
-            splade_scores = torch.log1p(torch.log1p(torch.relu(mlm_logits)))
-        else:
-            raise ValueError("activation_function must be either 'relu' or 'log1p_relu'")
+        # Unsqueeze attention_mask to be [batch_size, seq_length, 1] for broadcasting
+        attention_mask_expanded = attention_mask.unsqueeze(-1).to(mlm_logits.dtype)
+
+        batch_size, seq_len, vocab_s = mlm_logits.shape
+        device = mlm_logits.device
 
-        # Pool across sequence length dimension
+        # Initialize pooled scores based on pooling strategy
         if self.pooling_strategy == "max":
-            pooled_scores = torch.max(splade_scores, dim=1)[0]  # shape: batch_size, vocab_size
-        else:  # sum
-            pooled_scores = torch.sum(splade_scores, dim=1)  # shape: batch_size, vocab_size
+            pooled_scores = torch.full((batch_size, vocab_s), float("-inf"), dtype=mlm_logits.dtype, device=device)
+        elif self.pooling_strategy == "sum":
+            pooled_scores = torch.zeros((batch_size, vocab_s), dtype=mlm_logits.dtype, device=device)
+        else:
+            raise ValueError(f"Unsupported pooling_strategy: {self.pooling_strategy}")
+
+        # Process in chunks if chunk_size is set, otherwise process the entire sequence at once
+        chunk_size = seq_len if self.chunk_size is None else self.chunk_size
```

Comment:
```suggestion
        chunk_size = seq_len if (self.chunk_size is None or self.chunk_size <= 0) else self.chunk_size
```
I think this is safest to avoid an issue with the `range`